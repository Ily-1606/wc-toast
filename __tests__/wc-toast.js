import { createToaster } from '../test/utils.js';

describe('wc-toast', () => {
  afterEach(() => {
    document.body.innerHTML = '';
  });

  it('should add wc-toast with position=top-center to the DOM', () => {
    createToaster();
    const toaster = document.querySelector('wc-toast');

    expect(toaster.getAttribute('position')).toBe('top-center');
    expect(toaster.getAttribute('aria-live')).toBe('polite');
    expect(toaster.getAttribute('role')).toBe('status');

    expect(toaster.style.getPropertyValue('--wc-toast-factor')).toBe('1');
    expect(toaster.style.getPropertyValue('--wc-toast-position')).toBe('center');
    expect(toaster.style.getPropertyValue('--wc-toast-direction')).toBe('column-reverse');
    expect(toaster.outerHTML).toMatchSnapshot();
  });

  it('should add wc-toast with position=top-left to the DOM', () => {
    const position = 'top-left';
    createToaster(position);
    const toaster = document.querySelector('wc-toast');

    expect(toaster.style.getPropertyValue('--wc-toast-factor')).toBe('1');
    expect(toaster.style.getPropertyValue('--wc-toast-position')).toBe('flex-start');
    expect(toaster.style.getPropertyValue('--wc-toast-direction')).toBe('column-reverse');
    expect(toaster.outerHTML).toMatchSnapshot();
  });

  it('should add wc-toast with position=top-right to the DOM', () => {
    const position = 'top-right';
    createToaster(position);
    const toaster = document.querySelector('wc-toast');

    expect(toaster.style.getPropertyValue('--wc-toast-factor')).toBe('1');
    expect(toaster.style.getPropertyValue('--wc-toast-position')).toBe('flex-end');
    expect(toaster.style.getPropertyValue('--wc-toast-direction')).toBe('column-reverse');
    expect(toaster.outerHTML).toMatchSnapshot();
  });

  it('should add wc-toast with position=bottom-center to the DOM', () => {
    const position = 'bottom-center';
    createToaster(position);
    const toaster = document.querySelector('wc-toast');

    expect(toaster.style.getPropertyValue('--wc-toast-factor')).toBe('-1');
    expect(toaster.style.getPropertyValue('--wc-toast-position')).toBe('center');
    expect(toaster.style.getPropertyValue('--wc-toast-direction')).toBe('column');
    expect(toaster.outerHTML).toMatchSnapshot();
  });

  it('should add wc-toast with position=bottom-left to the DOM', () => {
    const position = 'bottom-left';
    createToaster(position);
    const toaster = document.querySelector('wc-toast');

    expect(toaster.style.getPropertyValue('--wc-toast-factor')).toBe('-1');
    expect(toaster.style.getPropertyValue('--wc-toast-position')).toBe('flex-start');
    expect(toaster.style.getPropertyValue('--wc-toast-direction')).toBe('column');
    expect(toaster.outerHTML).toMatchSnapshot();
  });

  it('should add wc-toast with position=bottom-right to the DOM', () => {
    const position = 'bottom-right';
    createToaster(position);
    const toaster = document.querySelector('wc-toast');

    expect(toaster.style.getPropertyValue('--wc-toast-factor')).toBe('-1');
    expect(toaster.style.getPropertyValue('--wc-toast-position')).toBe('flex-end');
    expect(toaster.style.getPropertyValue('--wc-toast-direction')).toBe('column');
    expect(toaster.outerHTML).toMatchSnapshot();
  });
});
