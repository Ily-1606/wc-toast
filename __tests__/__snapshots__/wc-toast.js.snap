// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wc-toast should add wc-toast with position=bottom-center to the DOM 1`] = `"<wc-toast position=\\"bottom-center\\" role=\\"status\\" aria-live=\\"polite\\" style=\\"--wc-toast-factor:-1; --wc-toast-position:center; --wc-toast-direction:column;\\"></wc-toast>"`;

exports[`wc-toast should add wc-toast with position=bottom-left to the DOM 1`] = `"<wc-toast position=\\"bottom-left\\" role=\\"status\\" aria-live=\\"polite\\" style=\\"--wc-toast-factor:-1; --wc-toast-position:flex-start; --wc-toast-direction:column;\\"></wc-toast>"`;

exports[`wc-toast should add wc-toast with position=bottom-right to the DOM 1`] = `"<wc-toast position=\\"bottom-right\\" role=\\"status\\" aria-live=\\"polite\\" style=\\"--wc-toast-factor:-1; --wc-toast-position:flex-end; --wc-toast-direction:column;\\"></wc-toast>"`;

exports[`wc-toast should add wc-toast with position=top-center to the DOM 1`] = `"<wc-toast position=\\"top-center\\" role=\\"status\\" aria-live=\\"polite\\" style=\\"--wc-toast-factor:1; --wc-toast-position:center; --wc-toast-direction:column-reverse;\\"></wc-toast>"`;

exports[`wc-toast should add wc-toast with position=top-left to the DOM 1`] = `"<wc-toast position=\\"top-left\\" role=\\"status\\" aria-live=\\"polite\\" style=\\"--wc-toast-factor:1; --wc-toast-position:flex-start; --wc-toast-direction:column-reverse;\\"></wc-toast>"`;

exports[`wc-toast should add wc-toast with position=top-right to the DOM 1`] = `"<wc-toast position=\\"top-right\\" role=\\"status\\" aria-live=\\"polite\\" style=\\"--wc-toast-factor:1; --wc-toast-position:flex-end; --wc-toast-direction:column-reverse;\\"></wc-toast>"`;
